### DISCUSSION TITLE
Package typings not following package's folder hierachy?

### DISCUSSION THREAD

**Discussion by haleksandre** [sveltejs/kit#6267] [Q&A]
Hi, 

When packaging I noticed the declaration files matches the original folder hierarchy `package/src/lib/[folders] but not the package's `package/[folders]`?

Here's an example of what I mean:
```
|--package
|    |-- components
|    |    |-- button.svelte
|    |
|    |-- src
|         |-- lib
|              |-- components
|                   |-- button.svelte.d.ts
|-- src
     |-- lib
         |-- components
               |-- button.svelte
```

Shouldn't the `package/src/lib/components/button.svelte.d.ts` be generated like so `package/components/button.svelte.d.ts` so that both the ts/svelte file & it's delaration are side by side?

**Comment by vekunz** [comment]
I "fixed" this for myself with this little node script. But indeed this should be fixed directly in `@sveltejs/package`.
```js
import * as fs from 'fs/promises';

await processDir('');
await fs.rm('package/lib', {recursive: true});

async function processDir(path) {
  const dir = await fs.opendir(`package/lib${path}`);

  for await (const item of dir) {
    if (item.isDirectory()) {
      await processDir(`${path}/${item.name}`);
    } else if (item.isFile() && !item.name.includes('.spec.') && !item.name.includes('.stories.')) {
      const srcPath = `package/lib${path}/${item.name}`;
      const destPath = `package${path}/${item.name}`;
      await fs.cp(srcPath, destPath, {force: true});
    }
  }
}
```

**Comment by dummdidumm** [comment]
That's strange, it should work as you expect. Is this reproducible with a simple hello world component example?
